src/build_system.jl

```diff
@@ -19,100 +17,109 @@ function build_system(
     category::Type{<:SystemCategory},
     name::String,
     print_stat::Bool = false;
+    force_build::Bool = false,
+    assign_new_uuids::Bool = false,
+    skip_serialization::Bool = false,
+    system_catalog::SystemCatalog = SystemCatalog(SYSTEM_CATALOG),
     kwargs...,
 )
-    system_catalog = get(kwargs, :system_catalog, SystemCatalog(SYSTEM_CATALOG))
     sys_descriptor = get_system_descriptor(category, system_catalog, name)
-    return _build_system(category, name, sys_descriptor, print_stat; kwargs...)
-end
+    sys_kwargs = filter_kwargs(; kwargs...)
+    sys_args = Dict{Symbol, Any}(k => v for (k, v) in sys_kwargs)
+    sys_keys = keys(sys_args)
 
-function build_system(
-    category::Type{T},
-    name::String,
-    print_stat::Bool = false;
-    kwargs...,
-) where {T <: Union{PSIDTestSystems, PSIDSystems}}
-    system_catalog = get(kwargs, :system_catalog, SystemCatalog(SYSTEM_CATALOG))
-    sys_descriptor = get_system_descriptor(category, system_catalog, name)
     psid_kwargs = check_kwargs_psid(; kwargs...)
-    if !isempty(psid_kwargs)
-        kwarg_type = first(values(psid_kwargs))
-        name = "$(name)_$kwarg_type"
-    end
-    return _build_system(
-        category,
-        name,
-        sys_descriptor,
-        print_stat;
-        add_forecasts = false,
-        kwargs...,
+    psid_args = Dict{Symbol, Any}(k => v for (k, v) in psid_kwargs)
+    psid_keys = keys(psid_args)
+
+    case_keys = keys(get_supported_arguments_dict(sys_descriptor))
```

Comment:
Seems like you need `get_supported_argument_names(d::SystemDescriptor) = [x.name for x in d.supported_arguments]`